src/sardana/macroserver/macros/standard.py

```diff
@@ -1049,3 +1049,41 @@ def run(self, ScanFilePath_list, ScanID):
 
         for postNewfileHook in self.getHooks('post-newfile'):
             postNewfileHook()
+
+
+class plotselect(Macro):
+    """select channels for plotting in the active measurement group"""
+    param_def = [
+          ['channel',
+           [['channel', Type.ExpChannel, 'None', ""], {'min': 0}],
+           None,
+           "List of channels to plot"],
+     ]
+
+    def run(self, channel):
+        try:
+            active_meas_grp = self.getEnv('ActiveMntGrp')
+        except UnknownEnv:
```

Comment:
In the `scanstats` macro in #880 I suggested to raise the exception to prevent the eventual subsequent macros in the sequence to proceed. I would recommend the same here. In this case it may be less critical because this macro does not produce any results that other macros could expect but still I'm in favor of ending with the exception.

I will make the necessary change and use the macro's `env` class member to indicate that the `ActiveMntGrp` is mandatory.

Reply:
Done in 336bb46

Reply:
I am pretty fine with that.